File: src/LIT.Smabu.UseCases/Invoices/UpdateInvoiceItem/UpdateInvoiceItemCommand.cs
﻿using LIT.Smabu.Domain.Common;
using LIT.Smabu.Domain.InvoiceAggregate;
using LIT.Smabu.Shared.Interfaces;

namespace LIT.Smabu.UseCases.Invoices.UpdateInvoiceItem
{
    public record UpdateInvoiceItemCommand : ICommand<InvoiceItemDTO>
    {
        public required InvoiceItemId Id { get; set; }
        public required InvoiceId InvoiceId { get; set; }
        public required string Details { get; set; }
        public required Quantity Quantity { get; set; }
        public required decimal UnitPrice { get; set; }
    }
}

File: src/lit.smabu.react/src/types/domain/update-invoice-item-command.ts
import { CatalogItemId } from './catalog-item-id';
import { InvoiceId } from './invoice-id';
import { InvoiceItemId } from './invoice-item-id';
import { Quantity } from './quantity';

export interface UpdateInvoiceItemCommand {
  invoiceItemId?: InvoiceItemId;
  invoiceId?: InvoiceId;
  details?: string | null;
  quantity?: Quantity;
  unitPrice?: number;
  catalogItemId?: CatalogItemId;
}
